GeosCore/CMakeLists.txt

```diff
@@ -80,13 +81,13 @@ add_library(GeosCore
   ocean_mercury_mod.F90
   olson_landmap_mod.F90
   pbl_mix_mod.F90
+  photolysis_mod.F90
   planeflight_mod.F90
   pops_mod.F90
   RnPbBe_mod.F90
   rpmares_mod.F90
   seasalt_mod.F90
   set_global_ch4_mod.F90
-  set_prof_o3.F90
   sfcvmr_mod.F90
   sulfate_mod.F90
   tagged_co_mod.F90
```

Comment:
I will add some descriptive comments as I review, as this PR has a lot of files (mostly for my own benefit).  This update to `CMakeLists.txt` adds the renamed module for photolysis and the interface module to the list of compilation.